client_cli/src/main.rs

```diff
@@ -892,6 +906,130 @@ mod asset {
             Ok(())
         }
     }
+
+    #[derive(StructOpt, Debug)]
+    pub struct SetStore {
+        /// AssetId for the Store asset (in form of `asset##account@domain_name')
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+        /// The value to be associated with the key
+        #[structopt(long)]
+        pub value: String,
+        /// Interpret the `value` as a JSON string
+        #[structopt(long = "json", requires = "value")]
+        pub is_json: bool,
+    }
+
+    impl RunArgs for SetStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self {
+                asset_id,
+                key,
+                value,
+                is_json,
+            } = self;
+
+            let v: Value = if is_json {
+                serde_json::from_str(&value)?
+            } else {
+                value.parse()?
+            };
+
+            let set = SetKeyValue::asset(asset_id, key, v);
+            submit([set], UnlimitedMetadata::default(), context)?;
+            Ok(())
+        }
+    }
+    #[derive(StructOpt, Debug)]
+    pub struct RemoveStore {
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+    }
+
+    impl RunArgs for RemoveStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { asset_id, key } = self;
+            let remove = RemoveKeyValue::asset(asset_id, key);
+            submit([remove], UnlimitedMetadata::default(), context)?;
+            Ok(())
+        }
+    }
+
+    #[derive(StructOpt, Debug)]
+    pub struct GetStore {
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        // The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+    }
+
+    impl RunArgs for GetStore {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { asset_id, key } = self;
+            let value = get_store(asset_id, &key, context)?;
+            context.print_data(&value)?;
+            Ok(())
+        }
+    }
+
+    #[derive(StructOpt, Debug)]
+    pub struct UpdateStore {
+        /// AssetId for the Store asset (in form of `asset##account@domain_name')
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+        /// The new value to be associated with the key
+        #[structopt(long)]
+        pub value: String,
+        /// Interpret the `value` as a JSON string
+        #[structopt(long = "json", requires = "value")]
+        pub is_json: bool,
+    }
+
+    impl RunArgs for UpdateStore {
```

Comment:
Imo update is redundant (get, set, del is enough), but we can wait for other opinions.

Reply:
I agree. There are two mutations: set and remove. CLI should reflect the API of Iroha as close as possible, imo.